### ISSUE TITLE
Add methods to pause and resume real-time synchronization

### ISSUE THREAD

**Issue by hackerwins** [yorkie-team/yorkie#469]
**What would you like to be added**:

Add methods to pause and resume real-time synchronization.

When attaching a document, we can choose whether or not to synchronize the document in real-time.

```ts
client.attach(doc); // isRealtimeSync: true (default: true)
client.attach(doc, true); // isRealtimeSync: false
```

Let's add methods(pause, resume) to change whether or not to synchronize already attached documents in real-time.

```ts
client.pause(doc); // isRealtimeSync: false
client.resume(doc); // isRealtimeSync: true
```

**Why is this needed**:

In some editors, local [text composition](https://developer.mozilla.org/en-US/docs/Web/API/Element/compositionstart_event) of [IME](https://developer.mozilla.org/en-US/docs/Glossary/Input_method_editor) is canceled when the content is changed by remote operation. This causes problems in Korean, where words are written using text composition.

Users can call `pause` in the text composition to keep it from receiving remote operations.

**Comment by easylogic** [comment]
The concept I have in mind is to delete from the `WatchDocument` target on pause() and `watchDocument` again on resume().

Is this concept correct?

**Comment by easylogic** [comment]
resume, pause status image 

<img width="1314" alt="스크린샷 2023-02-23 오후 9 20 05" src="https://user-images.githubusercontent.com/591983/220904059-11d3e3e0-8f78-4188-9315-2254c2d0f236.png">

**Comment by humdrum** [comment]
The isRealtimeSync flag is related to Presence too.

https://github.com/yorkie-team/yorkie-js-sdk/blob/5f71ea6680eb7eaf578a0b7583745c70ba6a6bec/src/core/client.ts#L557-L564

So. When the client paused the document, the presence update is also paused.

**Comment by hackerwins** [comment]
@humdrum That's a good point. I missed considering Presence. 🤔

@easylogic

We might need to add a `pullChanges` buffer like a `pushChanges` buffer.

- Local Editing(pushChanges buffer): https://github.com/yorkie-team/yorkie/blob/main/design/document-editing.md#local-editing
- Remote Editing: https://github.com/yorkie-team/yorkie/blob/main/design/document-editing.md#remote-editing

**Comment by hackerwins** [comment]
The issue about Korean IME will be addressed in #482.

Apart from the IME issue, We will provide pause/resume API that allows developers to change synchronization modes between real-time and manual.

**Comment by hackerwins** [comment]
https://github.com/yorkie-team/yorkie-js-sdk/pull/482
